Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/DefaultChartDataCollection.java

```diff
@@ -156,186 +130,87 @@ public DefaultChartDataCollection(APMLog apmLog) {
-                    if (i == numOfUnits) current = last;
-                    else current += avgWTUnit;
-                    avgWTMap.put(current, 0);
-                }
-            }
-        }
+        DoubleArrayList avgWTKeys = initDoubleSeriesMap(avgWaitTimes, avgWTMap);
+
         /**
          * For max waiting time
          */
         TreeSortedMap<Double, Integer> maxWTMap = new TreeSortedMap<>();
-        double maxWTUnit = 0;
-        if (allMaxWaitingTimes.size() > 0 ) {
-            if (interval > 0) {
-                Collections.sort(allMaxWaitingTimes);
-                int numOfUnits = MAX_UNIT;
-                double first = allMaxWaitingTimes.get(0);
-                double last = allMaxWaitingTimes.get(allMaxWaitingTimes.size() - 1);
-                maxWTUnit = (last - first) / numOfUnits;
-                double current = first - maxWTUnit;
-                for (int i = 0; i <= numOfUnits; i++) {
-                    if (i == numOfUnits) current = last;
-                    else current += maxWTUnit;
-                    maxWTMap.put(current, 0);
-                }
-            }
-        }
+        DoubleArrayList maxWTKeys = initDoubleSeriesMap(maxWaitTimes, maxWTMap);
+
         /**
          * For case utilization
          */
         TreeSortedMap<Double, Integer> caseUtilMap = new TreeSortedMap<>();
-        double caseUtilUnit = 0;
-        if (allCaseUtils.size() > 0 ) {
-            if (interval > 0) {
-                Collections.sort(allCaseUtils);
-                int numOfUnits = (MAX_UNIT / 2);
-                double first = allCaseUtils.get(0);
-                double last = allCaseUtils.get(allCaseUtils.size() - 1);
-                caseUtilUnit = (last - first) / numOfUnits;
-                double current = first - caseUtilUnit;
-                for (int i = 0; i < numOfUnits; i++) {
-                    current += caseUtilUnit;
-                    caseUtilMap.put(current, 0);
-                }
-                caseUtilMap.put(last, 0);
-                caseUtilMap.put(last, 0);
-            }
-        }
+        DoubleArrayList caseUtilKeys = initDoubleSeriesMap(caseUtils, caseUtilMap);
 
         /**
          * Loop (2):
          * - Fill in ...
          */
+        DoubleArrayList[] keysArray = new DoubleArrayList[] {
+                caseDurKeys,
+                ttlPTKeys,
+                avgPTKeys,
+                maxPTKeys,
+                ttlWTKeys,
+                avgWTKeys,
+                maxWTKeys,
+                caseUtilKeys
+        };
+
+        List<TreeSortedMap<Double, Integer>> seriesMapsList = new ArrayList<>();
+        seriesMapsList.add(caseDurMap);
+        seriesMapsList.add(ttlPTMap);
+        seriesMapsList.add(avgPTMap);
+        seriesMapsList.add(maxPTMap);
+        seriesMapsList.add(ttlWTMap);
+        seriesMapsList.add(avgWTMap);
+        seriesMapsList.add(maxWTMap);
+        seriesMapsList.add(caseUtilMap);
```

Comment:
You might be able to do this more easily as:
`List<TreeSortedMap<Double, Integer>> seriesMapsList = Arrays.asList(caseDurMap, ttlPTMap, avgPTMap, maxPTMap, ttlWTMap, avgWTMap, maxWTMap, caseUtilMap);`